src/basic_recipes/barplot.jl

```diff
@@ -59,12 +59,12 @@ function AbstractPlotting.plot!(p::BarPlot)
             x_unique = unique(filter(isfinite, x))
             
             if length(x_unique) == 1
-                width = 1.0
+                width = 1 - 2x_gap
```

Comment:
I'm respecting the previous implementation here, but maybe it could be easier to have `x_gap` default to `0.2` (rather than `0.1`) and just do `width = 1 - x_gap`. Somehow it feels natural for `x_gap` to measure the gap between bars rather than the "half gap".